tools/ClassGen/NodeBuilder.h

```diff
@@ -30,6 +30,73 @@
 #include <vector>
 
 class Builder;
+class NodeBuilder;
+
+class EnumBuilder {
+  /// The name of the enum.
+  std::string name_;
+  /// The fully qualified name of the enum (i.e. with all enclosing namespaces).
+  std::string fullyQualifiedName_;
+  /// The containing namespaces for the enum.
+  std::vector<std::string> namespaces_;
+  /// The enumeration names, values, and their corresponding docstrings.
+  std::vector<std::tuple<std::string, int, std::string>> enumerations_;
+  /// CPP file stream.
+  std::ofstream &cStream;
+  /// Specifies if this enum is backend specific. If true, a definition
+  /// will not be generated.
+  bool isBackendSpecific_{false};
+  /// Documentation string printed with the class definition.
+  std::string docstring_;
+  /// An integer value that is guaranteed to be unused by the enum at all times.
+  int unusedValue_{0};
+
+public:
+  /// Constructor.
+  EnumBuilder(std::ofstream &C, const std::string &name,
+              bool isBackendSpecific);
+
+  /// Set the documentation string. Each line will be prepended with "/// ".
+  EnumBuilder &setDocstring(const std::string &docstring) {
+    docstring_ = docstring;
+    return *this;
+  }
+
+  /// Add a new enumeration to the enumeration.
+  EnumBuilder &addEnumeration(const std::string &name,
```

Comment:
I'd call it `addCase` or `addEnumerator`. Enumeration is usually used to refer to the whole enum.

Reply:
Okay, I've fixed this locally and it will be included with my next push of major revisions.